ImperatorToCK3/Source/CK3Outputter/outTitles.cpp

```diff
@@ -5,6 +5,61 @@
 #include "../commonItems/CommonFunctions.h"
 
 
+void CK3::outputTitleHistory(const std::string& outputModName, const std::shared_ptr<Title>& title, std::ofstream& outputStream, std::set<std::string>& alreadyOutputted)
+{
+	outputStream << title->getName() << " = {\n";
+	if (title->holder == "0") outputStream << "\t1.1.1 = { holder = 0 }\n";
+	else
+	{
+		outputStream << "\t867.1.1 = {\n";
+		if (title->getDeFactoLiege()) outputStream << "\t\tliege = " << title->getDeFactoLiege()->getName() << "\n";
+		outputStream << "\t\tholder = " << title->holder << "\n";
+		outputStream << "\t}\n";
+	}
+	outputStream << "}\n";
+	alreadyOutputted.insert(title->getName());
```

Comment:
This would fit better in the caller that here:
1. Tracking which titles have been output isn't really part of outputting a single title.
2. It reduces the number of in/out parameters.

Reply:
That's right, why didn't I think of that ,😄

Reply:
That's why everyone needs code reviews. Even the smartest people can overlook things.

And as for the rest of us...